src/htm/engine/Output.cpp

```diff
@@ -114,7 +114,7 @@ void Output::addLink(const std::shared_ptr<Link> link) {
   links_.insert(link);
 }
 
-void Output::removeLink(std::shared_ptr<Link> link) {
+void Output::removeLink(const std::shared_ptr<Link>& link) {
```

Comment:
OT: reference to a pointer to a (L)link? :smiley:

Reply:
The same reason you pass in a const reference to a string.  In both places it copies very little but there is a copy and a reference count bump if you pass by value.